gramps/plugins/db/dbapi/dbapi.py

```diff
@@ -81,6 +82,26 @@ class DBAPI(DbGeneric):
     def _initialize(self, directory, username, password):
         raise NotImplementedError
 
+    def use_json_data(self):
+        """
+        A DBAPI level method for testing if the
+        database supports JSON access.
+        """
+        # Check if json_data exists on metadata as a proxy to see
+        # if the database has been converted to use JSON data
+        return self.dbapi.column_exists("metadata", "json_data")
+
+    def upgrade_table_for_json_data(self, table_name):
+        """
+        A DBAPI level method for upgrading the given table
+        adding a json_data column.
+        """
+        try:
+            self.dbapi.execute("ALTER TABLE %s ADD COLUMN json_data TEXT;" % table_name)
+            self.dbapi.commit()
+        except Exception:
+            pass
```

Comment:
Do you want to pass on *all* exceptions or just those thrown because the column json_data already exists?

Reply:
Yeah, that's a good point. I rewrote this without using try/except.